utils/check-incremental

```diff
@@ -8,84 +8,91 @@
 #
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
-#
-# ----------------------------------------------------------------------------
-#
-# This is a wrapper for the swift compiler.
-# It invokes the compiler multiple times and checks if the output object file
-# is only written a single time.
-# The main purpose of the check is to ensure that the compiler is
-# deterministic.
-#
-# ----------------------------------------------------------------------------
+"""
+Wrapper for the Swift compiler.
+
+It invokes the compiler multiple times, and checks if the output object
+file is only written once.  This checks that the compiler is deterministic.
+"""
 
+from __future__ import print_function
+
+import argparse
 import os
 import subprocess
 import sys
 
 
-def main():
-    verbose = False
-    num_iterations = 4
+def get_arguments():
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(__doc__)
 
-    write_obj_file = False
-    next_arg_is_output = False
-    compare_time = True
-    output_file = None
+    parser.add_argument('-o', '--output', metavar='OUTPUT_FILE', required=True,
+                        help='output file to compare')
+    parser.add_argument('-c', required=True,
+                        action='store_true', dest='write_obj_file',
+                        help='write_obj_file')  # @@AWLC: wut?
```

Comment:
I wasn’t really sure what this argument was supposed to be for – the original code exits if this flag is missing, but doesn’t do anything else with it.

I’m guessing this is a flag that gets passed to the Swift compiler somewhere?